File: Daily Challanges/2026-07-July/2026-07-16.py
def pig_latin(text):
    vowels = "aeiou"

    def convert(word):
        first_upper = word[0].isupper()
        lower_word = word.lower()

        if lower_word[0] in vowels:
            result = lower_word + "way"
        else:
            index = 0
            while index < len(lower_word) and lower_word[index] not in vowels:
                index += 1
            result = lower_word[index:] + lower_word[:index] + "ay"

        if first_upper:
            result = result[0].upper() + result[1:]

        return result

    return " ".join(convert(word) for word in text.split())


print(pig_latin("universe"))
print(pig_latin("hello"))
print(pig_latin("hello universe"))
print(pig_latin("Hello universe"))
print(pig_latin("Pig Latin is fun"))
print(pig_latin("The quick brown fox jumped over the lazy dog"))

File: Daily Challanges/2026-07-July/2026-07-16.js
function pigLatin(text) {
  const vowels = "aeiou";

  function convert(word) {
    const firstUpper = word[0] === word[0].toUpperCase();
    const lowerWord = word.toLowerCase();
    let result;

    if (vowels.includes(lowerWord[0])) {
      result = lowerWord + "way";
    } else {
      let index = 0;
      while (index < lowerWord.length && !vowels.includes(lowerWord[index])) {
        index++;
      }
      result = lowerWord.slice(index) + lowerWord.slice(0, index) + "ay";
    }

    if (firstUpper) {
      result = result[0].toUpperCase() + result.slice(1);
    }

    return result;
  }

  return text.split(" ").map(convert).join(" ");
}

console.log(pigLatin("universe"));
console.log(pigLatin("hello"));
console.log(pigLatin("hello universe"));
console.log(pigLatin("Hello universe"));
console.log(pigLatin("Pig Latin is fun"));
console.log(pigLatin("The quick brown fox jumped over the lazy dog"));
